### DISCUSSION TITLE
[Bug? / Feature Request] Allow mouse input in log view

### DISCUSSION THREAD

**Discussion by tenpaiyomi** [F1bonacc1/process-compose#32] [Ideas]
Loving the app, it's a potential solution for a lot of headaches I've had with some other process managers, so thanks for that.  One thing that would be great, if possible, is to support being able to copy text within the log view, as often I need to copy little snippets out for debugging or sharing.

**Comment by F1bonacc1** [comment]
Hi @tenpaiyomi 

Good idea!
Are you specifically talking about mouse select and copy or maybe something like "save the log buffer to a file", triggered by a shortcut?

**Reply by tenpaiyomi** [reply]
Mouse select and copy, specifically.  Everything else works well enough that I can scroll through and find what I want, I just don't have the ability to select it.  I like the tmux approach of selecting text and it auto copies, so I wind up using that often and it was the first thing I attempted.

**Reply by F1bonacc1** [reply]
@tenpaiyomi 

Can you please check if using the shift key while selecting works for you?

**Reply by tenpaiyomi** [reply]
@F1bonacc1 it does, however there are 2 issues with that, which I see.

First, it copies the entirety of the selection of rows, including the borders created by process-compose and (in my case, since I am using tmux and have a separate pane currently) anything in an extraneous pane.

Second, it provides no way for me to copy lines of text which may be off screen.  In the case of tmux, to use that again, when I am dragging text and I get to the edge of the pane, it will start scrolling the pane and continuing the selection.

**Reply by F1bonacc1** [reply]
@tenpaiyomi 
I will check if it's possible to override the default mouse selection behavior in the third party I am using.

**Comment by F1bonacc1** [comment]
After I checked the third party that I use, I didn't find a convenient API to do this properly.
I contacted the developer and after some back a forth, he might consider adding it to the library:
https://github.com/rivo/tview/issues/774

In the meanwhile, I came up with an alternative solution. Once you want to select something, you hit a hotkey, `process-compose` will then open a dialog with your log buffer in it. I will be scrollable and selectable, but I still need to figure out how well the clipboard copying will behave, specifically all the cross-platform issues.
It's not the optimal solution, but maybe this can work until the library is updated.

What do you think?

**Reply by tenpaiyomi** [reply]
I'm certainly willing to give it a try!  Right now I've been using `overmind`, but I love the approach that `process-compose` has for treating the processes in a way that allows me to set dependencies and the like, which helps when it comes to starting / stopping processes in the proper order, so anything that helps me to be able to swap over is a win in my book!

**Comment by F1bonacc1** [comment]
Wow, this feature took me to unexpected places... :)
It appears that there is no multi-platform clipboard package in go, that doesn't require additional installations.
So after scouring the internet for a few days, I eventually made a new one from bits and pieces that I found.
https://github.com/f1bonacc1/glippy
I was able to test it in Linux and Windows, if you have access to OSX, please let me know if everything works as expected.

In any case, give it a try in v0.29.0 and let me know if it works for you.
Any kind of feedback is more than welcome.
Cheers
